sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -2584,6 +2584,14 @@ abstract class SQLQuerySuiteBase extends QueryTest with SQLTestUtils with TestHi
       }
     }
   }
+
+  test("SPARK-33474: Support TypeConstructed partition spec value") {
+    withTable("t") {
+      sql("CREATE TABLE t(name STRING) PARTITIONED BY (part DATE) STORED AS ORC")
```

Comment:
Please add more tests for types and add tests in `DDLParserSuite`.

Reply:
btw, why did you use `stored as orc` here for this test?

Reply:
> btw, why did you use `stored as orc` explicitly for this test?

removed

Reply:
> Please add more tests for the other types and add tests in `DDLParserSuite`.

Both updated